src/models/notifyRecipient/notifyRecipient.ts

```diff
@@ -0,0 +1,67 @@
+import { publish } from "../../services/sns";
+import { getSpmTemplateInput, SpmPayload } from "./notification";
+import { config } from "../../config";
+import { getLogger } from "../../common/logger";
+import { TestData } from "../healthCert";
+
+const { trace } = getLogger("src/models/notifyRecipient");
+
+interface NotifyRecipientProps {
+  url: string;
+  qrCode: string;
+  nric?: string;
+  passportNumber: string;
+  testData: TestData[];
+  validFrom: string;
+}
+
+/* eslint-disable @typescript-eslint/camelcase */
+export const notifyRecipient = async ({
+  url,
+  qrCode,
+  nric,
+  passportNumber,
+  testData,
+  validFrom
+}: NotifyRecipientProps) => {
+  if (!config.notification.enabled) {
+    trace(
+      `Notification is disabled: Enable it by setting NOTIFICATION_ENABLED=true`
+    );
+    return;
+  }
+
+  if (!nric) {
```

Comment:
Possible to move this SPM notification template to its own repo as a library?
SNS is fine.

SPM specific details should not be included by default as the actual use case of SNS is to allow consuming by different digital wallet systems.

SPM is just one of the consumer today for us, we could include different wallet systems to be notified, as well as for external parties to implement their own wallets

Reply:
Implemented in #18